### PR TITLE
refactor webApponLinuxDeploy

### PR DESCRIPTION
**Pull request by Eskibear**
* add Copyright
* add eventListener to `RunProcessHandler.java`
* create data structure `ImageSetting` (e.g. ACR setting)
* rename `RemoteRunPanel`
* add presenter & view for webAppOnLinux conf dialog

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/ui/SettingPanel.java [hunk 1]
```diff
@@ -31,9 +70,17 @@
...
+    private WebAppOnLinuxDeployPresenter<SettingPanel> webAppOnLinuxDeployPresenter;
+
+    public SettingPanel() {
+        webAppOnLinuxDeployPresenter = new WebAppOnLinuxDeployPresenter<>();
+        webAppOnLinuxDeployPresenter.onAttachView(this);
 
-    public RemoteRunPanel() {
-        AzureMvpModel.getInstance();
         btnRefreshList.addActionListener(event -> onRefreshButtonSelection());
         comboResourceGroup.setRenderer(new ListCellRendererWrapper<Object>() {
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/ui/SettingPanel.java]: Change Object to Subscription

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java [hunk 2]
```diff
@@ -46,15 +67,71 @@ public void deployWebApp() {
...
+        WebApp app = null;
+        Azure azure = AuthMethodManager.getInstance().getAzureManager().getAzure(sid);
+
+        // handle AppServicePlan/ResourceGroup setting
+        WebApp.DefinitionStages.WithDockerContainerImage withDockerContainerImage;
+        if (profile.isResourceGroupExisting()) {
+            withDockerContainerImage = azure.webApps().define(profile.getWebAppName())
+                    .withRegion(Region.findByLabelOrName(profile.getRegionName()))
+                    .withExistingResourceGroup(profile.getResourceGroupName()) // existing RG
+                    .withNewLinuxPlan(new PricingTier(profile.getPricingTierSkuTier(), profile.getPricingTierSkuSize
```
Comment (jdneo) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java]: new line after comma

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/ui/SettingPanel.java [hunk 3]
```diff
@@ -238,15 +230,15 @@ private void loadCombosInfo(String activeWebAppId) {
             for (ResourceEx<SiteInner> si : AzureWebAppMvpModel.getInstance().listWebAppsOnLinuxBySubscriptionId(sb
                     .subscriptionId(), false)) {
                 comboWebApps.addItem(si);
-                if (activeWebAppId != null && si.getResource().id() == activeWebAppId) {
+                if (activeWebAppId != null && si.getResource().id().equals(activeWebAppId)) {
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/ui/SettingPanel.java]: We can leverage Comparing.equal(a, b) here to check the null issue

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/ui/SettingPanel.java [hunk 4]
```diff
@@ -205,4 +248,60 @@ private void loadCombosInfo(String activeWebAppId) {
...
+            System.out.println("table selected");
+            if (cachedList != null) {
+                selectedWebApp = cachedList.get(webAppTable.getSelectedRow());
+            }
+        });
+
+        AnActionButton refreshAction = new AnActionButton("Refresh", AllIcons.Actions.Refresh) {
+            @Override
+            public void actionPerformed(AnActionEvent anActionEvent) {
+                webAppOnLinuxDeployPresenter.onRefresh(true);
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/ui/SettingPanel.java]: remove the param in onRefresh

**Pull request merged**